### PR TITLE
Partitioning individual doc updates

### PR DESCRIPTION
**Pull request by Amruta-Ranade**

### CODE REVIEW

File: v2.0/configure-replication-zones.md [hunk 1]
```diff
@@ -306,6 +310,28 @@ Alternately, you can pass the YAML content via the standard input:
...
+{{site.data.alerts.callout_info}}This is an <a href="enterprise-licensing.html">enterprise-only</a> feature.{{site.data.alerts.end}}
+
+To [control replication for table partitions](partitioning.html#replication-zones), create a YAML file defining only the values you want to change (other values will not be affected), and use the `cockroach zone set <database.table.partition> -f <file.yaml>` command with appropriate flags:
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cat > australia.zone.yml
+~~~
+
+~~~ shell
```
Comment (jseldess) [inline v2.0/configure-replication-zones.md]: nit: You can remove `shell` for blocks showing a response.

Comment (Amruta-Ranade) [reply v2.0/configure-replication-zones.md]: Those are not responses. They are inputs to the file.

File: v2.0/create-table.md [hunk 2]
```diff
@@ -67,11 +67,18 @@ Parameter | Description
...
 `table_constraint` | An optional, comma-separated list of [table-level constraints](constraints.html). Constraint names must be unique within the table but can have the same name as columns, column families, or indexes.
 `opt_interleave` | You can potentially optimize query performance by [interleaving tables](interleave-in-parent.html), which changes how CockroachDB stores your data.
+`opt_partition_by` | <span class="version-tag">New in v2.0</span> An [enterprise-only](enterprise-licensing.html) option that lets you define table partitions at the row level. You can define table partitions by list or by range. See [Define Table Partitions](partitioning.html) for more information.
```
[the remark this answers is not in the document]
Comment (Amruta-Ranade) [reply v2.0/create-table.md]: Done.

File: v2.0/partition-by.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+{{site.data.alerts.callout_info}}<a href="partitioning.html">Defining table partitions</a> is an <a href="enterprise-licensing.html">enterprise-only</a> feature.{{site.data.alerts.end}}
+
+<div id="toc"></div>
+
+## Primary Key Requirements
+
+The [primary key required for partitioning](partitioning.html#partition-using-primary-key) is different from the conventional primary key: The unique identifier in the primary key requires to be prefixed with all columns you want to partition and subpartition the table on, in the order in which you want to nest your subpartitions.
+
+As of CockroachDB v2.0, you cannot alter the primary key after it has been defined while creating the table. [creating the table](create-table.html#create-a-table-with-partitions-new-in-v2-0). If the primary key in your existing table does not meet the requirements, you won't be able to use the `ALTER TABLE` statement to define partitions or subpartitions on the existing table. Allow users to alter the primary key after table creation is on the roadmap for CockroachDB v2.1.
```
[the remark this answers is not in the document]
Comment (Amruta-Ranade) [reply v2.0/partition-by.md]: Done.

File: v2.0/partition-by.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+`list_partitions` | Name of list partition followed by the list of values to be included in the partition.
+`range_partitions` | Name of range partition followed by the range of values to be included in the partition.
+
+## Required Privileges
+
+The user must have the `CREATE` [privilege](privileges.html) on the table.
+
+## Examples
+
+### Alter an Existing Table to Define Partitions by List
```
[the remark this answers is not in the document]
Comment (Amruta-Ranade) [reply v2.0/partition-by.md]: Done.

File: v2.0/partition-by.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+{% include copy-clipboard.html %}
+~~~ sql
+> ALTER TABLE students_by_list PARTITION BY LIST (country)
+      (PARTITION north_america VALUES IN ('CA','US'),
+      PARTITION australia VALUES IN ('AU','NZ'),
+      PARTITION DEFAULT VALUES IN (default));
+~~~
+
+### Alter an Existing Table to Define Partitions by Range
```
[the remark this answers is not in the document]
Comment (Amruta-Ranade) [reply v2.0/partition-by.md]: Done.

File: v2.0/partition-by.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+Suppose we have an another existing table named `students_by_range` and the primary key of the table is defined as `(expected_graduation_date, id)`. We can define partitions on the table by range:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> ALTER TABLE students_by_range PARTITION BY RANGE (expected_graduation_date)
+      (PARTITION graduated VALUES FROM (MINVALUE) TO ('2017-08-15'),
+      PARTITION current VALUES FROM ('2017-08-15') TO (MAXVALUE));
+~~~
+
+### Alter an Existing Table to Define Subpartitions
```
[the remark this answers is not in the document]
Comment (Amruta-Ranade) [reply v2.0/partition-by.md]: Done.

File: v2.0/partition-by.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+~~~
+
+### Unpartition a Table
+
+{% include copy-clipboard.html %}
+~~~ sql
+> ALTER TABLE students PARTITION BY NOTHING;
+~~~
+
+## See Also
```
[the remark this answers is not in the document]
Comment (Amruta-Ranade) [reply v2.0/partition-by.md]: Done.

File: v2.0/partitioning.md [hunk 8]
```diff
@@ -507,7 +507,7 @@ Time: 11.586626ms
 
 ### Repartition a Table
 
-Consider the partitioned table of students of RoachLearn. Suppose the table has been partitioned on range to store the current students on fast and expensive storage devices (example: SSD) and store the data of the graduated students on slower, cheaper storage devices(example: HDD). Now suppose we want to change the date after which the students will be considered current to `2018-08-15`. We can achieve this by using the `ALTER TABLE` command.
+Consider the partitioned table of students of RoachLearn. Suppose the table has been partitioned on range to store the current students on fast and expensive storage devices (example: SSD) and store the data of the graduated students on slower, cheaper storage devices(example: HDD). Now suppose we want to change the date after which the students will be considered current to `2018-08-15`. We can achieve this by using the [`ALTER TABLE`](alter-table.html) command.
```
[the remark this answers is not in the document]
Comment (Amruta-Ranade) [reply v2.0/partitioning.md]: Done.

File: v2.0/partitioning.md [hunk 9]
```diff
@@ -518,7 +518,7 @@ Consider the partitioned table of students of RoachLearn. Suppose the table has
 
 ### Unpartition a Table
 
-You can remove the partitions on a table by using the `PARTITION BY NOTHING` syntax with the `ALTER TABLE` command:
+You can remove the partitions on a table by using the `PARTITION BY NOTHING` syntax with the [`ALTER TABLE`](alter-table.html) command:
```
[the remark this answers is not in the document]
Comment (Amruta-Ranade) [reply v2.0/partitioning.md]: Done.

File: v2.0/configure-replication-zones.md (hunk 1 above)
Comment (jseldess) [reply v2.0/configure-replication-zones.md]: Ah, I see. I didn't understand what was happening with `cat >`.  We don't it this way in other zone config examples, but it's fine for now. We can revisit when we hash out more details style conventions. @lhirata, this is another topic for consideration (creating files).

**Comment by Amruta-Ranade** [comment]
@jseldess @lhirata @rmloveland Yes, that's a new thing I found out recently: If you want to view a file, `cat <filename>` suffices; if you want to edit a file you need `cat > <filename>`.

File: v2.0/configure-replication-zones.md (hunk 1 above)
Comment (Amruta-Ranade) [reply v2.0/configure-replication-zones.md]: @jseldess @lhirata @rmloveland Yes, that's a new thing I found out (the hard way) recently: If you want to view a file, `cat <filename>` suffices; if you want to edit a file you need `cat > <filename>`.

**Pull request merged**